test/Identity.FunctionalTests/ManagementTests.cs

```diff
@@ -53,6 +57,78 @@ public ManagementTests(ServerFactory serverFactory)
             await UserStories.ConfirmEmailAsync(email, client);
         }
 
+        [Fact]
+        public async Task CanChangePassword()
+        {
+            // Arrange
+            var principals = new List<ClaimsPrincipal>();
+            var server = ServerFactory.CreateServer(builder =>
+                builder.ConfigureTestServices(s => s.SetupGetUserClaimsPrincipal(user => principals.Add(user), IdentityConstants.ApplicationScheme)));
+
+            var client = ServerFactory.CreateDefaultClient(server);
+            var newClient = ServerFactory.CreateDefaultClient(server);
+
+            var userName = $"{Guid.NewGuid()}@example.com";
+            var password = "!Test.Password1";
+
+            var index = await UserStories.RegisterNewUserAsync(client, userName, password);
+
+            // Act 1
+            var changedPassword = await UserStories.ChangePasswordAsync(index, "!Test.Password1", "!Test.Password2");
+
+            // Assert 1
+            // RefreshSignIn generates a new security stamp claim
+            Assert.NotEqual(GetSecurityStampClaimValue(principals[0]), GetSecurityStampClaimValue(principals[1]));
+
+            // Act 2
+            await UserStories.LoginExistingUserAsync(newClient, userName, "!Test.Password2");
+
+            // Assert 2
+            // Signing in again with a different client uses the same security stamp claim
+            Assert.Equal(GetSecurityStampClaimValue(principals[1]), GetSecurityStampClaimValue(principals[2]));
+        }
+
+        [Fact]
+        public async Task CanSetPasswordWithExternalLogin()
+        {
+            // Arrange
+            var principals = new List<ClaimsPrincipal>();
+            var server = ServerFactory.CreateServer(builder =>
+                builder.ConfigureTestServices(s => s.SetupTestThirdPartyLogin()
+                .SetupGetUserClaimsPrincipal(user => principals.Add(user), IdentityConstants.ApplicationScheme)));
+
+            var client = ServerFactory.CreateDefaultClient(server);
+            var newClient = ServerFactory.CreateDefaultClient(server);
+
+            var guid = Guid.NewGuid();
+            var userName = $"{guid}";
+            var email = $"{guid}@example.com";
+
+            // Act 1
+            var index = await UserStories.RegisterNewUserWithSocialLoginAsync(client, userName, email);
+            index = await UserStories.LoginWithSocialLoginAsync(newClient, userName);
+
+            // Assert 1
+            Assert.NotNull(GetAuthenticationMethodIdentifierClaimValue(principals[1]));
+
+            // Act 2
+            var changedPassword = await UserStories.SetPasswordAsync(index, "!Test.Password2");
+
+            // Assert 2
+            // RefreshSignIn uses the same AuthenticationMethod claim value
+            Assert.Equal(GetAuthenticationMethodIdentifierClaimValue(principals[1]), GetAuthenticationMethodIdentifierClaimValue(principals[2]));
+        }
```

Comment:
I had to sign in after registering to get this claim. @HaoK Is this sufficient? I will add tests for other scenarios tomorrow.

Reply:
Yeah that looks fine

minor nit: maybe cleaner if you refactor the helper method to a single one like this: `AssertClaimsEqual(principals[1], principals[2], "AspNet.Identity.SecurityStamp"`)

Reply:
There will be another helper method `AssertClaimsNotEqual` but it is a lot cleaner. Thanks